### PR TITLE
[board-server] Read CORS config from environment variables instead of Firestore

### PR DESCRIPTION
**Pull request by timswanson-google**
Part of #2896

Switches from reading the Firestore table to determine CORS headers to using
the configured environment variables.

THIS IS A BREAKING CHANGE. All existing Board Server deployments will need to
have their environment config updated in order to continue functioning.

In the spirit of making the breaking change as minimal as possible, this PR does not delete the code that reads the (now redundant) Firestore table. That will be done in a separate cleanup step.

### CODE REVIEW

File: packages/board-server/src/server/cors.ts [hunk 1]
```diff
@@ -43,17 +44,18 @@ export const corsAll = (req: IncomingMessage, res: ServerResponse) => {
...
   const host = req.headers.host || "";
   const sameOrigin = origin.includes(host);
-  const headers = structuredClone(CORS_HEADERS);
+
   const isAllowed =
-    isLocalhost ||
-    sameOrigin ||
-    origin.length === 0 ||
-    CONFIG?.allow?.includes(origin);
+    sameOrigin || origin.length === 0 || allowedOrigins.has(origin);
```
Comment (aomarks) [inline packages/board-server/src/server/cors.ts:58]: I wonder if we can remove the `sameOrigin || origin.length === 0` part? Seems like the same origin case should be handled by `allowedOrigins`, and `origin.length` should actually not be allowed? (For another PR maybe, just something I noticed skimming this)

**Review by aomarks — APPROVED** [review]

**Comment by aomarks** [comment]
Nice!

**Comment by aomarks** [comment]
Looks there's a test failure.

File: packages/board-server/src/server/cors.ts (hunk 1 above)
Comment (timswanson-google) [reply packages/board-server/src/server/cors.ts:58]: Yeah, I was wondering about that myself. I might take a look and follow up on some of these conditions. I'm not sure what triggers them. (For example, when is origin empty?)

I want to do some debugging to see what these values are in practice.

**Comment by timswanson-google** [comment]
> Looks there's a test failure.

Fixed. Why doesn't this show up when I do `npm run build`?

**Comment by aomarks** [comment]
> > Looks there's a test failure.
> 
> Fixed. Why doesn't this show up when I do `npm run build`?

Looks like it's because `build` doesn't depend on `build:tests`, and `build:tests` is the only place we do type checkig here.

Side note: I'd suggest eventually removing esbuild and using only tsc instead, but that can be another fixup for later.

- tsc is the official typescript compiler. it does type checking, and produces the most likely to be correct .js output

- esbuild is a third party typescript compiler. it does NOT do type checking, and it's generally good at producing .js, but can be behind or slightly different to the official compiler. the reason people pick it is that its super fast (written in go instead of js), but I don't think we really care here, especially since we really want type checking anyway (and in fact, we are also calling tsc, but only for tests)

**Pull request merged**